_posts/2023-07-08-our-infrastructure.md

```diff
@@ -0,0 +1,261 @@
+It can also take over the web part completely, as it is just simple listing.
+But this too, future work.
+
+### CDN (releases, etc)
+
+[https://cdn.openttd.org/](https://cdn.openttd.org/) contains all our releases ever made.
+Every nightly (or at least the source), is available as far back as 2004.
+Release 0.1.1 can be found there.
+
+I hope obvious by now, this is hosted on Cloudflare R2, with a Cloudflare Worker in front of it.
+The Worker deals with GET requests, and make sure you see a nice directory listing when you visit the website.
+But is also deals with PUT requests, to add new files.
+
+These PUT requests need a `X-Signature`, to validate you are allowed to upload new files.
+This signature is signed with a private/public RSA-key, where only the services that need to upload files know the private part.
+
+For example, https://github.com/OpenTTD/OpenTTD has a GitHub Secret called `CDN_SIGNING_KEY`, which can only be used to upload files in `openttd-releases` and `openttd-nightlies`.
+This gives us a bit security that not just everyone can upload files, and prevents accident in scripts to delete or overwrite files; as this is simply impossible.
+
+### Other services (translator, DorpsGek, website, wiki)
+
+There are a ton of other services, like [translator](https://github.com/OpenTTD/eints), [DorpsGek](https://github.com/OpenTTD/DorpsGek) (our Discord / IRC bot), the [website](https://github.com/OpenTTD/website), [wiki](https://github.com/TrueBrain/TrueWiki), etc.
+
+None of them are really all that interesting.
+
+Translator and DorpsGek run in Nomad.
+The wiki does too, but Cloudflare caches as much as possible (as it is purely an HTTPS website; just very dynamic).
+The website is fully done in Cloudflare Pages: the GitHub workflow uploads a new site every day directly to Cloudflare Pages.
+
+### Other bits and pieces (docs, preview, redirects)
+
+So much more services we run to make things easier.
+
+We publish the documentation on [https://docs.openttd.org/](https://docs.openttd.org/).
+This is done when a new nightly is produced, and uploaded to a Cloudflage Pages.
+
+We also allow OpenTTD Pull Requests to be previewed.
+The WASM version of OpenTTD is also uploaded to Cloudflare Pages, but as a branch.
+And a Cloudflare Worker on [https://preview.openttd.org/](https://preview.openttd.org/) loads the right branch to serve you the latest preview.
+This in combination with [GitHub Deployments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment), gives you a nice "View Deployment" button to press on Pull Requests.
+
+Lastly, we have a ton of domains that simply redirect.
+[https://bugs.openttd.org/task/1](https://bugs.openttd.org/task/1) redirects to the GitHub issue, [https://forum.openttd.org/](https://forum.openttd.org/) points you to the subforum on tt-forums.net, etc.
```

Comment:
```suggestion
[https://bugs.openttd.org/task/1](https://bugs.openttd.org/task/1) redirects to the GitHub issue, [https://forum.openttd.org/](https://forum.openttd.org/) points you to the subforum on [tt-forums.net](tt-forums.net), etc.
```

Reply:
deliberately didn't make this a link :) But good catch :)